src/bridges/mean/MeanBridge.sol

```diff
@@ -57,6 +58,43 @@ contract MeanBridge is BridgeBase, Ownable2Step {
     // Note: we need to be able to receive ETH to deposit as WETH
     receive() external payable {}
 
+    /**
+     * @notice A function which will allow the user to create DCA positions on Mean Finance
+     * @param _inputAssetA - ETH or ERC20 token to deposit and start swapping
+     * @param _outputAssetA - ETH or ERC20 token to swap funds into
+     * @param _inputValue - Amount to deposit
+     * @param _interactionNonce - Unique identifier for this DeFi interaction
+     * @param _auxData - The amount of swaps, swap interval and wrappers encoded together
+     * @param _rollupBeneficiary - Address which receives subsidy if the call is eligible for it
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory _outputAssetB,
+        uint256 _inputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address _rollupBeneficiary
+    ) external payable override(BridgeBase) onlyRollup returns (uint256, uint256, bool) {
+        if (_inputAssetA.assetType != _outputAssetB.assetType || _inputAssetA.erc20Address != _outputAssetB.erc20Address) {
+            // We are making sure that input asset A = output asset B. We do this so that, if we need to, we can close the 
+            // position while it's still being swapped
+            revert ErrorLib.InvalidOutputB();
+        }
```

Comment:
Do we need to check if inputA != outputA? I know our smart contract on Ethereum mainnet will, but no sure how will that work for the Aztec transaction. Just double-checking with this Q tbh.

Reply:
I don't think it's necessary since the whole tx would fail when the hub reverts